client/my-sites/checkout/composite-checkout/wpcom/components/wp-checkout.js

```diff
@@ -127,8 +130,46 @@ export default function WPCheckout( {
 		setShouldShowContactDetailsValidationErrors,
 	] = useState( false );
 
+	const emailTakenLoginRedirectMessage = ( emailAddress ) => {
+		const redirectTo = '/checkout/no-site?cart=no-user';
+		const isNative = config.isEnabled( 'login/native-login-links' );
+		const loginUrl = login( { redirectTo, emailAddress, isNative } );
+		const loginRedirectMessage = translate(
+			'Choose a different email address. This one is not available. If this is you {{a}}log in now{{/a}}.',
+			{
+				components: {
+					a: <a href={ loginUrl } />,
+				},
+			}
+		);
+		return loginRedirectMessage;
+	};
+
 	const validateContactDetailsAndDisplayErrors = async () => {
 		debug( 'validating contact details with side effects' );
+		if ( isLoggedOutCart ) {
```

Comment:
I know this code isn't very DRY right now, but do we also need to use this block in `validateContactDetails` below? The difference between the functions is that `validateContactDetails` performs the validation and gets a response, but `validateContactDetailsAndDisplayErrors` also displays any errors. (`validateContactDetails` is used when we are checking existing saved contact details so that the contact details step might be skipped.)

Reply:
I've added `getSignupEmailValidationResult` to the `validateContactDetails ` block